ui/apps/platform/src/Containers/Integrations/utils/integrationUtils.ts

```diff
@@ -74,3 +76,70 @@ const getTimes = () => {
 };
 
 export const timesOfDay = getTimes();
+
+// Utilities for image integrations which can have either or both of two categories.
+
+// Categories alternatives correspond to mutually exclusive toggle group items.
+type CategoriesAlternatives<
+    Category0 extends ImageIntegrationCategory,
+    Category1 extends ImageIntegrationCategory
+> = [
+    [[category0: Category0]],
+    [[category1: Category1]],
+    // The alternative for both categories includes both orders.
```

Comment:
Aha! My initial reaction to this was why not use `ImageIntegrationCategory` directly, instead of creating two generic parameters, but this allows us to ensure that the third "alternatives" item contains both orderings at compile time.

It might be overkill, but if we want to ensure that these two items are always different as well we could update the type signatures in this file to:
```typescript
<
    Category0 extends ImageIntegrationCategory,
    Category1 extends Exclude<ImageIntegrationCategory, Category0>
>
```

Reply:
Super, thank you!